enterprise/internal/campaigns/search/changeset.go

```diff
@@ -0,0 +1,60 @@
+package search
+
+import (
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/sourcegraph/sourcegraph/enterprise/internal/campaigns"
+	"github.com/sourcegraph/sourcegraph/internal/search/query/syntax"
+)
+
+// ParseChangesetSearch parses the given search string into a set of options
+// that can be given to ListChangesets().
+//
+// At present, the only field that will be set in the options is TextSearch.
+// This will change in the future as we start to support field operators.
+func ParseChangesetSearch(search string) (*campaigns.ListChangesetsOpts, error) {
```

Comment:
Thought: if we introduce a separate type here (and translate it when passed to the store) then this package doesn't depend on our `enterprise/internal/campaigns` package and both the `resolver` and the `campaigns` package can depend on this one.

 ```suggestion
func ParseChangesetSearch(search string) (*ChangesetSearchOpts, error) {
```

But feel free to ignore, this is just a "hmm, couldn't we also..." thought I had while reading this.

Reply:
I'm inclined to leave it until it becomes a problem. It's technically possible, but once this supports field operators in #15783 et al then I'm not sure we're not going to end up with essentially a duplicate of `ListChangesetsOpts` anyway.

Reply:
Sounds good to me!